walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -21,82 +21,73 @@
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.Set;
+import java.util.Map;
+import java.util.HashMap;
 import com.google.sps.Events;
 import com.google.sps.MeetingRequest;
 import com.google.sps.TimeRange;
 
 public final class FindMeetingQuery {
   
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    
+    // duration of requested meeting
+    long duration = request.getDuration();
+    if (duration >= TimeRange.WHOLE_DAY.duration()) {
+        return Arrays.asList();
+    }
 
-    Collection<TimeRange> answer = Arrays.asList(TimeRange.WHOLE_DAY);
-
-    if (events == null) {
-        return answer; 
+    if (events.isEmpty()) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
     }
 
     //list of requested meeting attendees
     Collection<String> requestedAttendees = request.getAttendees();
-    if (requestedAttendees == null) {
-        return answer;
+    List<String> reqAttendees = new ArrayList<>();
+    reqAttendees.addAll(requestedAttendees);
+    if (reqAttendees == null) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
     }
 
     // list of optional meeting attendees
     Collection<String> optionalAttendees = request.getOptionalAttendees();
 
     //list of all events
-    List<Event> allEvents = new ArrayList<Event>();
+    List<Event> allEvents = new ArrayList<>();
     allEvents.addAll(events);
```

Comment:
You don't need to do this. Java Arraylist has a constructor that takes on a `Collection`

https://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html#ArrayList(java.util.Collection)

Reply:
```
List<Events> allEvents = new ArrayList<>(events);
```